### PR TITLE
Improves default parameter selection when dragging out snippets

### PR DESCRIPTION
**Pull request by darzu**
For example, dragging "onBlockPlaced" used to give (in TS & PY):
```
blocks.onBlockPlaced(0, function () { })
```
Now it gives:
```
blocks.onBlockPlaced(Block.Grass, function () { })
```

This improves both the typescript and python experience. This is also the foundation for doing the same improvement to our tab-complete. This is a huge usability improvement to our text experience, and it's especially important for our Python deliverable.

Lots of TODOs left in the code before it can be merged.

### CODE REVIEW

**Comment by jwunderl** [comment]
Seems like this will fix https://github.com/microsoft/pxt-arcade/issues/321 ?

File: pxtcompiler/emitter/service.ts [hunk 1]
```diff
@@ -1145,20 +1170,98 @@ namespace ts.pxtc.service {
...
+        console.log("attrs")
+        console.dir(attrs)
+        function getShadowSymbol(paramName: string): SymbolInfo | null {
+            let shadowBlock = (attrs._shadowOverrides || {})[paramName]
+            if (!shadowBlock)
+                return null
+            const shadowBlockMap: pxt.Map<string> = {
+                // shadow override -> qname
+                // TODO: determine this programatically somehow
+                "minecraftBlockField": "Block",
```
Comment (pelikhan) [inline pxtcompiler/emitter/service.ts]: add section in pxtarget.json

**Comment by pelikhan** [comment]
Is it possible to run some "getSnippet" as part of our test suite?

File: pxtcompiler/emitter/service.ts (hunk 1 above)
Comment (darzu) [reply pxtcompiler/emitter/service.ts]: We should be able to enumerate blocks and find ones with matching IDs

**Comment by darzu** [comment]
> Is it possible to run some "getSnippet" as part of our test suite?

Good question. What sort of test are you thinking? Syntactic matching against well-known "good" results?

**Comment by darzu** [comment]
> microsoft/pxt-arcade#321

Yes it should, thanks for the pointer. I assign it to myself.

**Comment by pelikhan** [comment]
Just a set of know snippets and recorded expected outputs.

**Comment by darzu** [comment]
> Just a set of know snippets and recorded expected outputs.

Currently nothing in the worker thread is tested or testable as far as I know.. I looked and I definitely want to do this but it's going to take some time (for me at least) to grok the structure and refactor things for testing. Pointers welcome. For now I'll not pull on that thread in this PR.

File: webapp/src/package.ts [hunk 2]
```diff
@@ -15,8 +16,8 @@ let extWeight: pxt.Map<number> = {
 }
 
 export function setupAppTarget(trgbundle: pxt.TargetBundle) {
-    //if (!trgbundle.appTheme) trgbundle.appTheme = {};
     pxt.setAppTarget(trgbundle)
+    compiler.initAppTarget(trgbundle)
```
Comment (pelikhan) [inline webapp/src/package.ts]: the app target is modified when a different board variant is loaded so you probably need this in more places.

Comment (darzu) [reply webapp/src/package.ts]: Thanks, I now also set the app target when the appTargetVariant changes

**Comment by pelikhan** [comment]
target bundle can be very large. best to pass the required data on the service call

**Comment by darzu** [comment]
> target bundle can be very large. best to pass the required data on the service call

Thanks, I changed it to do this. It works

**Pull request merged**